### PR TITLE
fix(rust,python) respect original series dtype when constructing `LitIter`

### LINKED ISSUE #9856: DateTime type lost after `unique()` inside GroupBy Context
### Checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import polars as pl
from datetime import date, timedelta

df = pl.DataFrame(
    {
        "key": ["A", "A", "A", "A"],
        "dates": [date(1970, 1, 1), date(1970, 1, 1), date(1970, 1, 2), date(1970, 1, 3)],
    }
)

# BROKEN
# output [86400000000, 86400000001, 86400000002]
df.groupby("key").agg(pl.col("dates").unique() + timedelta(days=1))



# WORKS
# output [1970-01-02, 1970-01-03, 1970-01-04]
df.groupby("key").agg((pl.col("dates") + timedelta(days=1)).unique())
```

### Issue description

When adding a timedelta to a deduplicated column inside a "GroupBy" context the type is lost and the output has type 'Int64'

### Expected behavior

The output type should be 'Date'

### Installed versions

<details>

```
--------Version info---------
Polars:              0.18.7
Index type:          UInt32
Platform:            macOS-13.3.1-arm64-arm-64bit
Python:              3.11.4 | packaged by conda-forge | (main, Jun 10 2023, 18:08:41) [Clang 15.0.7 ]

----Optional dependencies----
adbc_driver_sqlite:  <not installed>
connectorx:          <not installed>
deltalake:           <not installed>
fsspec:              <not installed>
matplotlib:          <not installed>
numpy:               1.25.0
pandas:              <not installed>
pyarrow:             <not installed>
pydantic:            <not installed>
sqlalchemy:          <not installed>
xlsx2csv:            <not installed>
xlsxwriter:          <not installed>
```

</details>

### PR DESCRIPTION
**Pull request by 0xbe7a**
closes #9856

### CODE REVIEW

File: polars/polars-lazy/src/physical_plan/expressions/group_iter.rs [hunk 1]
```diff
@@ -59,8 +60,14 @@ struct LitIter<'a> {
 }
 
 impl<'a> LitIter<'a> {
-    fn new(array: ArrayRef, len: usize, name: &str) -> Self {
-        let mut series_container = Box::pin(Series::try_from((name, array.clone())).unwrap());
+    fn new(array: ArrayRef, len: usize, logical: &DataType, name: &str) -> Self {
+        let mut series_container = Box::pin(
+            Series::try_from((name, array.clone()))
```
Comment (ritchie46) [inline polars/polars-lazy/src/physical_plan/expressions/group_iter.rs:65]: Can you use `Series::from_chunks_and_dtype_unchecked` here? That should `tag` `new` as unsafe.

**Comment by 0xbe7a** [comment]
Is there a reason `FlatIter` is still constructed `with cast().unwrap()` ?

File: polars/polars-lazy/src/physical_plan/expressions/group_iter.rs [hunk 2]
```diff
@@ -59,8 +63,15 @@ struct LitIter<'a> {
 }
 
 impl<'a> LitIter<'a> {
-    fn new(array: ArrayRef, len: usize, name: &str) -> Self {
-        let mut series_container = Box::pin(Series::try_from((name, array.clone())).unwrap());
+    /// # Safety
+    /// Caller must ensrue the goven `logical` dtype belongs to `array`.
```
Comment (0xbe7a) [inline polars/polars-lazy/src/physical_plan/expressions/group_iter.rs:67]:
```suggestion
    /// Caller must ensure the given `logical` dtype belongs to `array`.
```

**Comment by ritchie46** [comment]
> Is there a reason FlatIter is still constructed with cast(...).unwrap() ?

Nope, should do the same. Sorry that I pushed a few things. Want to get this in the new rust release. :)

**Pull request merged**